addons/lunch/models/lunch_alert.py

```diff
@@ -84,36 +87,92 @@ def _search_available_today(self, operator, value):
             ])
         ])
 
-    def _notify_chat(self):
-        records = self.search([('mode', '=', 'chat'), ('active', '=', True)])
-
-        today = fields.Date.today()
-        now = fields.Datetime.now()
-
-        for alert in records:
-            notification_to = now.astimezone(pytz.timezone(alert.tz)).replace(second=0, microsecond=0, tzinfo=None)
-            notification_from = notification_to - timedelta(minutes=5)
-            send_at = datetime.combine(fields.Date.today(),
-                float_to_time(alert.notification_time, alert.notification_moment))
-
-            if alert.available_today and send_at > notification_from and send_at <= notification_to:
-                order_domain = [('state', '!=', 'cancelled')]
 
-                if alert.location_ids.ids:
-                    order_domain = expression.AND([order_domain, [('user_id.last_lunch_location_id', 'in', alert.location_ids.ids)]])
-
-                if alert.recipients != 'everyone':
-                    weeks = 1
+    @api.depends('name', 'mode', 'notification_time', 'notification_moment', 'tz', 'active')
+    def _compute_cron_id(self):
+        for alarm in self:
+            alarm = alarm.with_context(tz=alarm.tz)
+
+            cron_required = (
+                alarm.active
+                and alarm.mode == 'chat'
+                and (not alarm.until or fields.Date.context_today(alarm) <= alarm.until)
+            )
+
+            now_tz = fields.Datetime.context_timestamp(alarm, datetime.utcnow())
+            sendat_tz = datetime.combine(date.today(), float_to_time(alarm.notification_time, alarm.notification_moment, alarm.tz))
```

Comment:
This combination is inconsistent: `date.today()` is today **in UTC**, while the time is **in a specific timezone**.